xbmc/interfaces/builtins/PlayerBuiltins.cpp

```diff
@@ -179,6 +179,23 @@ static int PlayerControl(const std::vector<std::string>& params)
       g_application.GetAppPlayer().SetTempo(playTempo);
     }
   }
+  else if (StringUtils::StartsWithNoCase(params[0], "tempo"))
+  {
+    if (params[0].size() == 5)
+      CLog::Log(LOGERROR, "PlayerControl(tempo(n)) called with no argument");
+    else if (params[0].size() < 8) // arg must be at least "(N)"
+      CLog::Log(LOGERROR, "PlayerControl(tempo(n)) called with invalid argument: \"%s\"",
+                params[0].substr(6).c_str());
+    else if (g_application.GetAppPlayer().SupportsTempo() &&
+             g_application.GetAppPlayer().IsPlaying() && !g_application.GetAppPlayer().IsPaused())
```

Comment:
As you are calling `g_application.GetAppPlayer()` at least once, it could make sense to assign the value to a variable and to reuse that value for the subsequent calls.

else
{
   player = g_application.GetAppPlayer();
   if (player.Supportstempo() && player....)
   {
      std::string strTempo = ...
      ...
   }
}

Reply:
updated this as well..